src/backend/distributed/commands/utility_hook.c

```diff
@@ -542,6 +544,16 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 		parsetree = pstmt->utilityStmt;
 		ops = GetDistributeObjectOps(parsetree);
 
+		/*
+		 * Preprocess and qualify steps can cause pg tests to fail because of the
+		 * unwanted citus related warnings or early error logs related to invalid address.
+		 * Therefore, we first check if all addresses in the given statement are valid.
+		 * Then, we do not execute qualify and preprocess if any address is invalid to
+		 * prevent before-mentioned citus related messages. PG will complain about the
+		 * invalid address, so we are safe to not execute qualify and preprocess.
```

Comment:
also, maybe note that for InvalidObjects, Postgres will throw error anyway, so we don't need to worry about anything further than standardProcess_Utility